src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,675 @@
+        TreeNode<IMenu> currentMenuTreeNode = menuTreeNode;
+        while (null != currentMenuTreeNode.getParent())
+        {
+            treeNodeMenuListMinusRoot.add(currentMenuTreeNode);
+            currentMenuTreeNode = currentMenuTreeNode.getParent();
+        }
+        sb.append('/').append(currentMenuTreeNode.getData().getMenuItemName());
+
+        Collections.reverse(treeNodeMenuListMinusRoot);
+        for (final TreeNode<IMenu> treeNode : treeNodeMenuListMinusRoot)
+        {
+            sb.append(' ').append(treeNode.getData().getMenuItemName());
+        }
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args) throws CommandException
+    {
+        final PermissionsChecker permissionChecker = getPermissionsChecker();
+        final ModuleContext moduleContext = getModuleContext();
+        execute(server, sender, args, permissionChecker, moduleContext);
+    }
+
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args, 
+            @NotNull final PermissionsChecker permissionsChecker, @NotNull final ModuleContext moduleContext) throws CommandException
+    {
+        final BlockPos pos = null;
+        final ParsingResult parsingResult = getTabCompletionsAndParsingHolders(getRoot(), server, sender, args, pos, moduleContext);
+        final TreeNode<IMenu> executionTreeNode = parsingResult.getExecutionTreeNode();
+        if (null == executionTreeNode)
+        {
+            throw new CommandException(getCommandUsage(sender, getRoot()));
+        }
+
+        final IMenu executionMenu = executionTreeNode.getData();
+        if (executionMenu.getMenuType().isNavigationMenu())
+        {
+            throw new CommandException(getCommandUsage(sender, executionTreeNode));
+        }
+
+        @NotNull final List<ActionArgument> executionActionArgumentList = parsingResult.getExecutionActionArgumentList();
+        // actionMenuState will be non-null at this point
+        @NotNull final ActionMenuState actionMenuState = parsingResult.getActionMenuState();
+        final ActionMenu actionMenu = actionMenuState.getActionMenu();
+
+        final String badArgument = parsingResult.getBadArgument();
+        throwCommandUsageExceptionIfRequiredArgumentsAreNotProvided(executionTreeNode, actionMenu, executionActionArgumentList, actionMenuState, badArgument, sender);
+
+        if (sender instanceof EntityPlayer)
+        {
+            final ForgePermissionNodes forgePermissionNode = actionMenu.getForgePermissionNode();
+
+            final EntityPlayer player = (EntityPlayer) sender;
+            if (!permissionsChecker.hasPermission(forgePermissionNode, player))
+            {
+                // TODO: Do something if permission check fails.
+                // But we don't have permissions set up yet.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)